### PR TITLE
test_other.py cleanups

### PR DESCRIPTION
**Pull request by froydnj**
The `temp_directory` changes are the interesting ones; the `test_single_file` change was thrown in for fun, and I can understand if you don't want it.  (There are other places that could use the `test_single_file` change, but I'm keeping it simple for the moment.)

### CODE REVIEW

**Comment by kripken** [comment]
Nice changes, thanks! I like both.

About the tempdir stuff, perhaps run those locally with `EM_TESTRUNNER_DETECT_TEMPFILE_LEAKS=1` in the env, just to be sure. (I don't remember why we don't run with that enabled by default, need to ask @juj, but his bots use it.)

**Comment by saschanaz** [comment]
Bad news: we still have that random `test_iostream_and_determinism` failure 😱

File: tests/test_other.py [hunk 1]
```diff
@@ -503,11 +494,7 @@ def check_makefile(configuration, dirname):
       ]
       for test_dir, output_file, cmake_args in cases:
         cmakelistsdir = path_from_root('tests', 'cmake', test_dir)
-        # Create a temp workspace folder
-        tempdirname = tempfile.mkdtemp(prefix='emscripten_test_' + self.__class__.__name__ + '_', dir=TEMP_DIR)
```
Comment (saschanaz) [inline tests/test_other.py]: Are we okay to throw these dynamic prefixes away?

**Comment by kripken** [comment]
> Bad news: we still have that random test_iostream_and_determinism failure 😱

:( Well, at least it's a lot more rare now than before. Here it turned up in -O1, so it's definitely not related to the stuff fixed before (which was -O3 only).

**Comment by saschanaz** [comment]
My previous repro was done with -O2, though. Hmm.

**Comment by froydnj** [comment]
I am also hitting some temporary file leaks with the suggested `EM_TESTRUNNER_DETECT_TEMPFILE_LEAKS=1`:

```
======================================================================
ERROR: test_binaryen_debug (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/froydnj/src/emscripten.git/tests/runner.py", line 233, in tearDown
    raise Exception('Test leaked ' + str(len(left_over_files)) + ' temporary files!')
Exception: Test leaked 1 temporary files!

======================================================================
ERROR: test_binaryen_opts (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/froydnj/src/emscripten.git/tests/runner.py", line 233, in tearDown
    raise Exception('Test leaked ' + str(len(left_over_files)) + ' temporary files!')
Exception: Test leaked 12 temporary files!

======================================================================
ERROR: test_no_nuthin (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/froydnj/src/emscripten.git/tests/runner.py", line 233, in tearDown
    raise Exception('Test leaked ' + str(len(left_over_files)) + ' temporary files!')
Exception: Test leaked 4 temporary files!

======================================================================
ERROR: test_no_nuthin_2 (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/froydnj/src/emscripten.git/tests/runner.py", line 233, in tearDown
    raise Exception('Test leaked ' + str(len(left_over_files)) + ' temporary files!')
Exception: Test leaked 2 temporary files!

======================================================================
ERROR: test_toolchain_profiler (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/froydnj/src/emscripten.git/tests/runner.py", line 233, in tearDown
    raise Exception('Test leaked ' + str(len(left_over_files)) + ' temporary files!')
Exception: Test leaked 4 temporary files!
```

File: tests/test_other.py (hunk 1 above)
Comment (froydnj) [reply tests/test_other.py]: It'd be pretty easy to modify `temp_directory` to include them; my initial class before I spotted `temp_directory` was setup like:

```python
  def __init__(self, test_runner, **kwargs):
    self.orig_cwd = None

    kwargs.setdefault('prefix', 'emscripten_test' + test_runner.__class__.__name__ + '_')
    kwargs.setdefault('dir', TEMP_DIR)
    self.name = mkdtemp(**kwargs)
```

so you'd do:

```python
with InTempDirectory(self):
    ....
```

**Comment by froydnj** [comment]
> I am also hitting some temporary file leaks with the suggested EM_TESTRUNNER_DETECT_TEMPFILE_LEAKS=1:

OTOH, none of these tests were modified by the patches, so perhaps this pull request is OK on that front.

**Comment by kripken** [comment]
Yeah, those sound unrelated. @juj did mention he saw some file leak errors on his bots, I guess they might be manually disabled there.

**Pull request merged**